common/debug-interface/src/node_debug_service.rs

```diff
@@ -29,21 +29,43 @@ impl NodeDebugService {
         // GET /events
         let events = warp::path("events").map(|| warp::reply::json(&json_log::pop_last_entries()));
 
-        // Post /log
-        let log = warp::post()
-            .and(warp::path("log"))
+        // Post /log/filter
+        let local_filter = {
+            let logger = logger.clone();
+
+            warp::path("filter")
+                // 16kb should be long enough for a filter
+                .and(warp::body::content_length_limit(1024 * 16))
+                .and(warp::body::bytes())
+                .map(move |bytes: bytes::Bytes| {
+                    if let (Some(logger), Ok(filter)) = (&logger, ::std::str::from_utf8(&bytes)) {
+                        info!(filter = filter, "Updating local logging filter");
+                        logger.set_filter(Filter::builder().parse(filter).build());
+                    }
+
+                    warp::reply::reply()
+                })
+        };
+
+        // Post /log/remote-filter
+        let remote_filter = warp::path("remote-filter")
             // 16kb should be long enough for a filter
             .and(warp::body::content_length_limit(1024 * 16))
             .and(warp::body::bytes())
             .map(move |bytes: bytes::Bytes| {
                 if let (Some(logger), Ok(filter)) = (&logger, ::std::str::from_utf8(&bytes)) {
-                    info!(filter = filter, "Updating logging filter");
-                    logger.set_filter(Filter::builder().parse(filter).build());
+                    info!(filter = filter, "Updating remote logging filter");
+                    logger.set_remote_filter(Filter::builder().parse(filter).build());
                 }
 
                 warp::reply::reply()
             });
 
+        // Post /log
+        let log = warp::post()
+            .and(warp::path("log"))
+            .and(local_filter.or(remote_filter));
```

Comment:
So this does both?

Reply:
`/log/filter` does the local and `/log/remote-filter` does the remote part

Reply:
Oh, I just wasn't sure what `filter.or(filter)` did both or what.